File: app/src/androidTest/java/com/shen/xi/android/tut/test/QingPageDriverTest.java
package com.shen.xi.android.tut.test;

import com.google.inject.Guice;
import com.google.inject.Injector;
import com.shen.xi.android.tut.sinablog.QingPageDriver;

import junit.framework.TestCase;

import java.io.InputStream;
import java.util.List;

public class QingPageDriverTest extends TestCase {

    private QingPageDriver mDrive = new QingPageDriver();

    @Override
    public void setUp() throws Exception {
        super.setUp();

        InputStream resourceAsStream = this.getClass().getClassLoader()
                .getResourceAsStream("assets/qingPage.html");
        mDrive.load(resourceAsStream);
        resourceAsStream.close();
        Injector injector = Guice.createInjector(new TestModule());
        injector.injectMembers(this);
    }

    @Override
    public void tearDown() throws Exception {

    }

    public void testLoadPage() {
        assertNotNull(mDrive.getImageUrlList());
    }

    public void testGetImageUrls() {
        List<String> imageUrlList = mDrive.getImageUrlList();
        assertEquals(5, imageUrlList.size());
        assertEquals("http://ww4.sinaimg.cn/mw600/e2830b12jw1eevfj15tf0j20tn18ggys.jpg",
                     imageUrlList.get(0));
    }
}

File: app/src/androidTest/scala/com/shen/xi/android/tut/test/QingPageDriverTest.scala
package com.shen.xi.android.tut.test

import com.google.inject.Guice
import com.google.inject.Injector
import com.shen.xi.android.tut.sinablog.QingPageDriver

import junit.framework.TestCase

import java.io.InputStream
import java.util.List

class QingPageDriverTest extends TestCase {

  import junit.framework.Assert._

  private val mDrive = new QingPageDriver()

  @throws[Exception]
  override def setUp() {
    super.setUp()

    val resourceAsStream = this.getClass
      .getClassLoader
      .getResourceAsStream("assets/qingPage.html")
    mDrive.load(resourceAsStream)
    resourceAsStream.close()
    val injector = Guice.createInjector(new TestModule())
    injector.injectMembers(this)
  }

  @throws[Exception]
  override def tearDown() {
  }

  def testLoadPage() {
    assertNotNull(mDrive.getImageUrlList)
  }

  def testGetImageUrls() {
    val imageUrlList = mDrive.getImageUrlList
    assertEquals(5, imageUrlList.size())
    assertEquals("http://ww4.sinaimg.cn/mw600/e2830b12jw1eevfj15tf0j20tn18ggys.jpg",
      imageUrlList.get(0))
  }
}
